### PR TITLE
Added tokens Bls12-381 to the Michelson Encoder

### LINKED ISSUE #888: Add a token Bls12-381 to the Michelson Encoder
**Is your feature request related to a problem and use case? Please describe.**

The type Bls12-381 was added in protocol 008, but is missing in the Michelson Encoder.

**Additional context**
https://tezos.gitlab.io/protocols/008_edo.html#bls12-381

### PR DESCRIPTION
**Pull request by roxaneletourneau**
closes #888 
Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: packages/taquito-michelson-encoder/src/tokens/bls12-381-fr.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    protected fac: TokenFactory
+  ) {
+    super(val, idx, fac);
+  }
+
+  private isValid(val: any): Bls12381frValidationError | null {
+    if (typeof val === 'string' && /^[0-9a-fA-F]*$/.test(val) && val.length % 2 === 0) {
+      return null;
+    } else {
+      return new Bls12381frValidationError(val, this, `Invalid bytes: ${val}`);
```
Comment (kjaiswal) [inline packages/taquito-michelson-encoder/src/tokens/bls12-381-fr.ts:26]: Would it provide more value if we add condition (out of above 3) on which validation failed to the returned error?

**Comment by kjaiswal** [comment]
What is the difference between 3 token types except their prim values. Is there a way we can avoid 3 files and make one that defines the 3 tokens?

**Comment by roxaneletourneau** [comment]
> What is the difference between 3 token types except their prim values. Is there a way we can avoid 3 files and make one that defines the 3 tokens?

These are 3 different Michelson types that I represented with 3 different classes that implement the token class. Can you give more detail on the solution you have in mind?

File: packages/taquito-michelson-encoder/src/tokens/bls12-381-fr.ts (hunk 1 above)
Comment (roxaneletourneau) [reply packages/taquito-michelson-encoder/src/tokens/bls12-381-fr.ts:26]: I'll make the change, thank you!

**Comment by kjaiswal** [comment]
I reckon each token type in Michelson has its own type file. So this should be okay.

**Review by kjaiswal — APPROVED** [review]

**Pull request merged**